Deprecated 'idleConnectionPoolSize' config option

lutovich opened the pull request:
Newly added 'maxConnectionPoolSize' option should be used instead.

lutovich committed:
Deprecated 'idleConnectionPoolSize' config option

Newly added 'maxConnectionPoolSize' option should be used instead.

zhenlineo commented on driver/src/main/java/org/neo4j/driver/v1/Config.java:
```diff
@@ -117,21 +129,26 @@ public boolean logLeakedSessions()
      * Max number of idle connections per URL for this driver.
+     *
      * @return the max number of connections
+     * @deprecated please use {@link #maxConnectionPoolSize()} instead.
```
I would suggest changing to `Please consider to use #maxConnectionPoolSize` and `#MaxConnectionLifetime` to replace this method instead.

Same applies for comments of all `MaxIdle` methods

lutovich replied:
Improved javadoc with your suggestion for `withMaxSessions`, `withMaxIdleSessions`, and `withMaxIdleConnections`.

zhenlineo commented on driver/src/main/java/org/neo4j/driver/v1/Config.java:
```diff
@@ -311,47 +327,54 @@ public ConfigBuilder withLeakedSessionsLogging()
         {
-            this.maxIdleConnectionPoolSize = size;
-            return this;
+            return withMaxConnectionPoolSize( size );
```
I was about to suggest to give a warning here, but then realized we might do not have a logger till this moment...

I feel we should not delegate maxIdle to max, as this config changes meaning already. 
I am wondering if it makes sense to just ignore the value we set to this method. What do you think of the following changes?

```
MaxPoolSize -> MaxConnectionPoolSize
MaxIdleConnections -> ignored
MaxIdleSessions -> ignored
```

ali-ince replied:
Since we do not have any idle connection checking in place on the async implementation, ignoring `MaxIdleConnections` and `MaxIdleSessions` config values makes more sense for me.

lutovich replied:
Thanks for pointing this out. I agree, will change

zhenlineo reviewed (commented):
Wonder if it makes sense to not mix maxIdle and max size, as they are kind of different in meanings?

lutovich committed:
Improved javadoc for couple Config methods

lutovich committed:
Ignore `maxIdleSessions` and `maxIdleConnections`

These two config settings will now be ignored and not delegate to
`maxConnectionPoolSize`. This feels better because logic of enforcing
"max idle" is now gone.

lutovich commented:
@zhenlineo, @ali-ince suggested config change implemented
